### PR TITLE
Wip 240 no index update on cassandra log init

### PR DESCRIPTION
**Pull request by krasserm**

### CODE REVIEW

File: eventuate-core/src/main/scala/com/rbmhtechnology/eventuate/log/EventLog.scala [hunk 1]
```diff
@@ -98,33 +106,43 @@ case class BatchReadResult(events: Seq[DurableEvent], to: Long) extends DurableE
...
  */
 private class PhysicalDeletionNotSupportedException extends UnsupportedOperationException
 
 /**
  * Event log storage provider interface (SPI).
+ *
+ * @tparam A Event log state type.
  */
-trait EventLogSPI { this: Actor =>
+trait EventLogSPI[A] { this: Actor =>
```
Comment (volkerstampa) [inline eventuate-core/src/main/scala/com/rbmhtechnology/eventuate/log/EventLog.scala]: Shouldnt we add here `<: EventLogState` as well?

File: eventuate-log-cassandra/src/main/scala/com/rbmhtechnology/eventuate/log/cassandra/CassandraEventLog.scala [hunk 2]
```diff
@@ -84,11 +86,27 @@ class CassandraEventLog(id: String) extends EventLog(id) {
...
 
+  private var index: ActorRef = null
   private var indexSequenceNr: Long = 0L
   private var updateCount: Long = 0L
 
+  override def recoverState: Future[CassandraEventLogState] = {
+    import services.readDispatcher
+    for {
+      dm <- deletedToStore.readDeletedToAsync.map(DeletionMetadata(_, Set.empty))
+      sc <- indexStore.readEventLogClockAsync
```
Comment (volkerstampa) [inline eventuate-log-cassandra/src/main/scala/com/rbmhtechnology/eventuate/log/cassandra/CassandraEventLog.scala]: Could it make sense to call this method `readEventLogClockSnapshotAsync`?

**Comment by volkerstampa** [comment]
LGTM

File: eventuate-core/src/main/scala/com/rbmhtechnology/eventuate/log/EventLog.scala (hunk 1 above)
Comment (krasserm) [inline eventuate-core/src/main/scala/com/rbmhtechnology/eventuate/log/EventLog.scala]: That's actually only a constraint of the abstract `EventLog` (from which all implementations extend).

File: eventuate-log-cassandra/src/main/scala/com/rbmhtechnology/eventuate/log/cassandra/CassandraEventLog.scala (hunk 2 above)
Comment (krasserm) [inline eventuate-log-cassandra/src/main/scala/com/rbmhtechnology/eventuate/log/cassandra/CassandraEventLog.scala]: Good idea, will do.

**Pull request merged**